plugins/azure-devops-backend/src/service/router.test.ts

```diff
@@ -0,0 +1,187 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ConfigReader } from '@backstage/config';
+import express from 'express';
+import request from 'supertest';
+import { AzureDevOpsApi } from '../api';
+import { createRouter } from './router';
+import { RepoBuild } from '../api/types';
+import { GitRepository } from 'azure-devops-node-api/interfaces/GitInterfaces';
+import {
+  Build,
+  BuildResult,
+  BuildStatus,
+} from 'azure-devops-node-api/interfaces/BuildInterfaces';
+
+describe('createRouter', () => {
+  let azureDevOpsApi: jest.Mocked<AzureDevOpsApi>;
+  let app: express.Express;
+
+  beforeAll(async () => {
+    azureDevOpsApi = {
+      getGitRepository: jest.fn(),
+      getBuildList: jest.fn(),
+      getRepoBuilds: jest.fn(),
+    } as any;
+    const router = await createRouter({
+      azureDevOpsApi,
+      logger: getVoidLogger(),
+      config: new ConfigReader({
+        azureDevOps: {
+          token: 'foo',
+          host: 'host.com',
+          organization: 'myOrg',
+          top: 5,
+        },
+      }),
+    });
+    app = express().use(router);
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('GET /health', () => {
+    it('returns ok', async () => {
+      const response = await request(app).get('/health');
+
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual({
+        status: 'Healthy',
+        details: 'All required config has been provided',
+      });
+    });
+  });
+
+  describe('GET /repository/:projectName/:repoName', () => {
+    it('fetches a single repository', async () => {
+      const gitRepository: GitRepository = {
+        id: 'af4ae3af-e747-4129-9bbc-d1329f6b0998',
+        name: 'myRepo',
+        url: 'https://host.com/repo',
+        defaultBranch: 'refs/heads/develop',
+        sshUrl: 'ssh://host.com/repo',
+        webUrl: 'https://host.com/webRepo',
+      };
+
+      azureDevOpsApi.getGitRepository.mockResolvedValueOnce(gitRepository);
+
+      const response = await request(app).get(
+        `/repository/${'myProject'}/${'myRepo'}`,
+      );
+
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual(gitRepository);
+    });
+  });
+
+  describe('GET /builds/:projectName/:repoId', () => {
+    it('fetches a list of builds', async () => {
+      const firstBuild: Build = {
+        id: 1,
+        buildNumber: 'Build-1',
+        status: BuildStatus.Completed,
+        result: BuildResult.Succeeded,
+        queueTime: '2020-09-12T06:10:23.9325232Z' as unknown as Date,
```

Comment:
this is also a place where you could leave out the type above so you don't have to do the `as` dance

Reply:
So what I've been running into when I try that is I remove all the types and the test themselves pass but there is a TS error doing for example this: `azureDevOpsApi.getBuildList.mockResolvedValueOnce(builds);`. Basically because `builds` isn't a `Build[]` anymore. Is there something else I should be trying?

Reply:
Yeah the reason there is, that's an _actual_ error, if I read this right. Saying `queueTime: '2020...'` is de facto wrong since the queue time is expected to be a Date, not a string. I may be missing something somewhere, but it sounds to me like you wanted to actually write `queueTime: Date.UTC...` something like here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC or to just do `queueTime: new Date()` if you don't care too much about the actual precise value